### PR TITLE
Automate image process to make images and thumbnails square

### PR DESCRIPTION
**Pull request by SpeedyD**
Credit for original code to https://github.com/itinerare / @itinerare 

Adds option in settings to automate thumbnail process, making images square and resizing them to thumbnail size.
Replaces the image cropper, but allows user to still upload their own thumbnail.

### CODE REVIEW

File: config/lorekeeper/settings.php [hunk 1]
```diff
@@ -164,6 +164,22 @@
...
+	| This feature will replace the thumbnail cropper as option at image uploads.
+	| It will automatically add transparent borders to the images to make them square,
+	| based on the bigger dimension (between width/height).
+	| Thumbnails will effectively be small previews of the full masterlist images.
+	| This feature will not replace the manual uploading of thumbnails.
+	|
+	| Simply change to "1" to enable, or keep at "0" to disable.
+	|
+    */
+	'masterlist_image_automation' => 0,
```
Comment (Draginraptor) [inline config/lorekeeper/settings.php:181]: ^ misaligned

File: resources/views/admin/masterlist/create_character.blade.php [hunk 2]
```diff
@@ -131,20 +131,36 @@
...
-            {!! Form::hidden('x0', null, ['id' => 'cropX0']) !!}
-            {!! Form::hidden('x1', null, ['id' => 'cropX1']) !!}
-            {!! Form::hidden('y0', null, ['id' => 'cropY0']) !!}
-            {!! Form::hidden('y1', null, ['id' => 'cropY1']) !!}
+			<img src="#" id="cropper" class="hide" />
+			{!! Form::hidden('x0', null, ['id' => 'cropX0']) !!}
+			{!! Form::hidden('x1', null, ['id' => 'cropX1']) !!}
+			{!! Form::hidden('y0', null, ['id' => 'cropY0']) !!}
+			{!! Form::hidden('y1', null, ['id' => 'cropY1']) !!}
         </div>
```
Comment (Draginraptor) [inline resources/views/admin/masterlist/create_character.blade.php:161]: ^ not aligned with `<div id="cropSelect">...`

File: resources/views/character/admin/_reupload_image_modal.blade.php [hunk 3]
```diff
@@ -3,20 +3,36 @@
         {!! Form::label('Character Image') !!} {!! add_help('This is the full masterlist image. Note that the image is not protected in any way, so take precautions to avoid art/design theft.') !!}
         <div>{!! Form::file('image', ['id' => 'mainImage']) !!}</div>
     </div>
+@if (Config::get('lorekeeper.settings.masterlist_image_automation') === 1)
     <div class="form-group">
         {!! Form::checkbox('use_cropper', 1, 1, ['class' => 'form-check-input', 'data-toggle' => 'toggle', 'id' => 'useCropper']) !!}
-        {!! Form::label('use_cropper', 'Use Image Cropper', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the image cropper (crop dimensions can be adjusted in the site code), or upload a custom thumbnail.') !!}
+		{!! Form::label('use_cropper', 'Use Thumbnail Automation', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the Thumbnail Automation, or upload a custom thumbnail.') !!}
```
Comment (Draginraptor) [inline resources/views/character/admin/_reupload_image_modal.blade.php:9]: ^ not aligned with `{!! Form::checkbox('use_cropper'...`

File: resources/views/character/admin/_reupload_image_modal.blade.php [hunk 4]
```diff
@@ -3,20 +3,36 @@
...
     </div>
     <div class="card mb-3" id="thumbnailCrop">
         <div class="card-body">
-            <div id="cropSelect">Select an image to use the thumbnail cropper.</div>
-            <img src="#" id="cropper" class="hide" />
+			<div id="cropSelect">By using this function, the thumbnail will be automatically generated from the full image.</div>
```
Comment (Draginraptor) [inline resources/views/character/admin/_reupload_image_modal.blade.php:13]: ^ misaligned with the below lines

File: resources/views/character/admin/_reupload_image_modal.blade.php [hunk 5]
```diff
@@ -3,20 +3,36 @@
...
+            {!! Form::hidden('x0', 1) !!}
+            {!! Form::hidden('x1', 1) !!}
+            {!! Form::hidden('y0', 1) !!}
+            {!! Form::hidden('y1', 1) !!}
+        </div>
+    </div>
+@else
+    <div class="form-group">
+        {!! Form::checkbox('use_cropper', 1, 1, ['class' => 'form-check-input', 'data-toggle' => 'toggle', 'id' => 'useCropper']) !!}
+		{!! Form::label('use_cropper', 'Use Image Cropper', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the image cropper (crop dimensions can be adjusted in the site code), or upload a custom thumbnail.') !!}
```
Comment (Draginraptor) [inline resources/views/character/admin/_reupload_image_modal.blade.php:23]: ^ misaligned with above line

File: resources/views/character/admin/upload_image.blade.php [hunk 6]
```diff
@@ -24,20 +24,36 @@
     {!! Form::label('Character Image') !!} {!! add_help('This is the full masterlist image. Note that the image is not protected in any way, so take precautions to avoid art/design theft.') !!}
     <div>{!! Form::file('image', ['id' => 'mainImage']) !!}</div>
 </div>
+@if (Config::get('lorekeeper.settings.masterlist_image_automation') === 1)
 <div class="form-group">
     {!! Form::checkbox('use_cropper', 1, 1, ['class' => 'form-check-input', 'data-toggle' => 'toggle', 'id' => 'useCropper']) !!}
-    {!! Form::label('use_cropper', 'Use Image Cropper', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the image cropper (crop dimensions can be adjusted in the site code), or upload a custom thumbnail.') !!}
+	{!! Form::label('use_cropper', 'Use Thumbnail Automation', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the Thumbnail Automation, or upload a custom thumbnail.') !!}
```
Comment (Draginraptor) [inline resources/views/character/admin/upload_image.blade.php:30]: ^ misaligned with above line

File: resources/views/character/admin/upload_image.blade.php [hunk 7]
```diff
@@ -24,20 +24,36 @@
...
 </div>
 <div class="card mb-3" id="thumbnailCrop">
     <div class="card-body">
-        <div id="cropSelect">Select an image to use the thumbnail cropper.</div>
-        <img src="#" id="cropper" class="hide" />
-        {!! Form::hidden('x0', null, ['id' => 'cropX0']) !!}
-        {!! Form::hidden('x1', null, ['id' => 'cropX1']) !!}
-        {!! Form::hidden('y0', null, ['id' => 'cropY0']) !!}
-        {!! Form::hidden('y1', null, ['id' => 'cropY1']) !!}
+		<div id="cropSelect">By using this function, the thumbnail will be automatically generated from the full image.</div>
```
Comment (Draginraptor) [inline resources/views/character/admin/upload_image.blade.php:34]: ^ misaligned with below lines

**Comment by itinerare** [comment]
(Just noting quickly that I am in fact cool with use of my code here)

File: resources/views/character/admin/upload_image.blade.php [hunk 8]
```diff
@@ -24,20 +24,36 @@
...
+        {!! Form::hidden('x0', 1) !!}
+        {!! Form::hidden('x1', 1) !!}
+        {!! Form::hidden('y0', 1) !!}
+        {!! Form::hidden('y1', 1) !!}
     </div>
 </div>
+@else
+<div class="form-group">
+    {!! Form::checkbox('use_cropper', 1, 1, ['class' => 'form-check-input', 'data-toggle' => 'toggle', 'id' => 'useCropper']) !!}
+		{!! Form::label('use_cropper', 'Use Image Cropper', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the image cropper (crop dimensions can be adjusted in the site code), or upload a custom thumbnail.') !!}
```
Comment (Draginraptor) [inline resources/views/character/admin/upload_image.blade.php:44]: ^ inconsistent spacing; I think it should be aligned with the second line

File: resources/views/character/admin/upload_image.blade.php [hunk 9]
```diff
@@ -24,20 +24,36 @@
...
+</div>
+<div class="card mb-3" id="thumbnailCrop">
+	<div class="card-body">
+		<div id="cropSelect">Select an image to use the thumbnail cropper.</div>
+		<img src="#" id="cropper" class="hide" />
+		{!! Form::hidden('x0', null, ['id' => 'cropX0']) !!}
+		{!! Form::hidden('x1', null, ['id' => 'cropX1']) !!}
+		{!! Form::hidden('y0', null, ['id' => 'cropY0']) !!}
+		{!! Form::hidden('y1', null, ['id' => 'cropY1']) !!}
+    </div>
```
Comment (Draginraptor) [inline resources/views/character/admin/upload_image.blade.php:54]: ^ slightly odd indentation of the card-body as a whole

File: resources/views/character/design/image.blade.php [hunk 10]
```diff
@@ -70,22 +70,36 @@
                 {!! Form::label('modify_thumbnail', 'Modify Thumbnail', ['class' => 'form-check-label ml-3']) !!} {!! add_help('Toggle this option to modify the thumbnail, otherwise only the credits will be saved.') !!}
             </div>
         @endif
+@if (Config::get('lorekeeper.settings.masterlist_image_automation') === 1)
         <div class="form-group">
             {!! Form::checkbox('use_cropper', 1, 1, ['class' => 'form-check-input', 'data-toggle' => 'toggle', 'id' => 'useCropper']) !!}
-            {!! Form::label('use_cropper', 'Use Image Cropper', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the image cropper (crop dimensions can be adjusted in the site code), or upload a custom thumbnail.') !!}
+			{!! Form::label('use_cropper', 'Use Thumbnail Automation', ['class' => 'form-check-label ml-3']) !!} {!! add_help('A thumbnail is required for the upload (used for the masterlist). You can use the Thumbnail Automation, or upload a custom thumbnail.') !!}
```
Comment (Draginraptor) [inline resources/views/character/design/image.blade.php:76]: ^ misaligned with above lines

**Review by Draginraptor — COMMENTED** [review]
Sorry for all the spam! I'll try to help fix up the formatting before merging.

**Comment by SpeedyD** [comment]
> Sorry for all the spam! I'll try to help fix up the formatting before merging.

Appreciated! Sorry for.. basically making a mess?? (Even though it's so clean and pristine on my end, thanks Git. X_x;..)

**Review by Draginraptor — APPROVED** [review]

**Review by itinerare — APPROVED** [review]

**Pull request merged**